Sources/BraveWallet/Panels/Signature Request/SignMessageRequestContainerView.swift

```diff
@@ -0,0 +1,95 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import SwiftUI
+import BraveStrings
+import BraveCore
+import DesignSystem
+
+/// View for displaying an array of `SignMessageRequest`s`
+struct SignMessageRequestContainerView: View {
+
+  var requests: [BraveWallet.SignMessageRequest]
+  @ObservedObject var keyringStore: KeyringStore
+  var cryptoStore: CryptoStore
+  @ObservedObject var networkStore: NetworkStore
+  var onDismiss: () -> Void
+
+  @State private var requestIndex: Int = 0
+  
+  /// A map between request index and a boolean value indicates this request message needs pilcrow formating
+  @State private var needPilcrowFormatted: [Int32: Bool] = [0: false]
+  /// A map between request index and a boolean value indicates this request message is displayed as
+  /// its original content
+  @State private var showOrignalMessage: [Int32: Bool] = [0: true]
+    
+  /// The current request
+  private var currentRequest: BraveWallet.SignMessageRequest {
+    requests[requestIndex]
+  }
+  
+  /// The account for the current request
+  private var currentRequestAccount: BraveWallet.AccountInfo {
+    keyringStore.allAccounts.first(where: { $0.address == currentRequest.accountId.address }) ?? keyringStore.selectedAccount
+  }
+  
+  /// The network for the current request
+  private var currentRequestNetwork: BraveWallet.NetworkInfo? {
+    networkStore.allChains.first(where: { $0.chainId == currentRequest.chainId })
+  }
+  
+  var body: some View {
+    Group {
+      if let ethSiweData = currentRequest.signData.ethSiweData {
+        SignInWithEthereumView(
+          account: currentRequestAccount,
+          originInfo: currentRequest.originInfo,
+          message: ethSiweData,
+          action: { approved in
+            cryptoStore.handleWebpageRequestResponse(.signMessage(approved: approved, id: currentRequest.id))
+            if requests.count <= 1 {
+              onDismiss()
+            }
+          }
```

Comment:
You can probably move this to a separate func in this View and use it for both conditions since they do the same